src/Database/Type/DateTimeType.php

```diff
@@ -157,7 +157,7 @@ public function marshal($value)
             if ($value === '' || $value === null || $value === false || $value === true) {
                 return null;
             }
-            if (is_numeric($value)) {
+            if (ctype_digit($value)) {
```

Comment:
Add $value=0392387900 to test case
This value is valid but can not be accept by ctype_digit()